### PR TITLE
Added migration from SteamlessTimes

### PR DESCRIPTION
**Pull request by ma3a**
Added migration from Steamless Times

### CODE REVIEW

**Review by Witherking25 — CHANGES_REQUESTED** [review]
Needs to be compatible with all permutations of SteamlessTimes, MetaDeck included

File: main.py [hunk 1]
```diff
@@ -67,10 +70,31 @@ async def get_overall_times(self):
         except Exception:
             logger.exception("Unhandled exception")
 
+    async def migrate_data_from_steamless_time(self, association):
+        try:
+            steamless_time_data_path = f"{decky_home}/settings/steamlesstimes.json"
```
[the remark this answers is not in the document]
Comment (ma3a) [reply main.py:75]: Yeah! I changed that yesterday, already trying to get `[steamlesstimes, metadeck].sort(last_update desc)[0]`, but haven't tested it yet, so didn't push. Will update it soon :)

**Comment by SilentException** [comment]
Tested but found some oddities. 
Got 4 errors (OK but still weird because nothing is missing or removed). I think the errors should be written to log file for later analysis).
I also still need to check what was actually written to sqlite but just wanted to leave a quick comment here. 
I believe this can be improved a bit.

**Comment by SilentException** [comment]
After checking the sqlite, the data was imported correctly.
As I mentioned, it would be cool to have more verbose import data logged, wouldn't hurt.
The 4 errors I had are puzzling because I see no reason why the IDs couldn't be matched. If it helps I narrowed it out to the point that all 4 were non-Steam games that I added in desktop mode using Add To Steam - Edge for xCloud, Prism Launcher launcher and Prism Launcher for Minecraft instances. I don't think this is uncommon. Other non steam games that wer linked correctly are emulation games added using Steam Rom Manager.

So I would say that the import is working correctly. What is really wonky is how the hours played data is shown. Not in plugin statistics but in home screen (recent games) and game info screens. It is really hit and miss when and how the hours played data is displayed. It might show for some games, then randomly hide, it might not work at all for other games, all kind of  issues there. This is unrelated to import but still something that should be addressed.

**Review by Witherking25 — APPROVED** [review]
I think this is good, I need to fix the display issues for the app page, but this doesn't affect that

**Comment by Witherking25** [comment]
> After checking the sqlite, the data was imported correctly.
> As I mentioned, it would be cool to have more verbose import data logged, wouldn't hurt.
> The 4 errors I had are puzzling because I see no reason why the IDs couldn't be matched. If it helps I narrowed it out to the point that all 4 were non-Steam games that I added in desktop mode using Add To Steam - Edge for xCloud, Prism Launcher launcher and Prism Launcher for Minecraft instances. I don't think this is uncommon. Other non steam games that wer linked correctly are emulation games added using Steam Rom Manager.
> 
> So I would say that the import is working correctly. What is really wonky is how the hours played data is shown. Not in plugin statistics but in home screen (recent games) and game info screens. It is really hit and miss when and how the hours played data is displayed. It might show for some games, then randomly hide, it might not work at all for other games, all kind of  issues there. This is unrelated to import but still something that should be addressed.

Were those games removed and then readded?

**Comment by ma3a** [comment]
> I think the errors should be written to log file for later analysis).

Yeah. already in latest changes

**Comment by SilentException** [comment]
> Were those games removed and then readded? If so, then the game id changed and it can't be imported

No, nothing was removed and readded.
I fixed the import by hand. Perhaps see how testing goes with other people. It should be noticeable quite quickly if there is something wrong.

**Comment by ma3a** [comment]
> that I added in desktop mode using Add To Steam - Edge for xCloud, Prism Launcher launcher and Prism Launcher for Minecraft instances. I don't think this is uncommon. Other non steam games that wer linked correctly are emulation games added using Steam Rom Manager.

Hmmm.. got the same things done(GamePass is just Edge passthrough, Zelda from Rom manager, and other from flatpacks and or GoG Store), and they have mappings:
```json 
{
"9773558945233764352": {"gameId": "2275584020", "name": "Fallout 2"},
"10751366219341758464": {"gameId": "2503247517", "name": "Zelda Wind Waker"},
"11764491675244691456": {"gameId": "2739134169", "name": "Prism Launcher"},
"15665255220571537408": {"gameId": "3647351456", "name": "EmulationStation-DE-x64_SteamDeck"},
"15840941622318596096": {"gameId": "3688256634", "name": "OpenMW Launcher"},
"16354849427246022656": {"gameId": "3807910119", "name": "The Legend of Zelda: Breath of the Wild"},
"17485176874140172288": {"gameId": "4071084986", "name": "Fallout: New Vegas Ultimate Edition"},
"17860396981034156032": {"gameId": "4158447725", "name": "GamePass"}
}
```

**Comment by SilentException** [comment]
Yeah, strange. 

So I looked at this file on my SD: /home/deck/.local/share/Steam/userdata/137789989/760/screenshots.vdf
In there there are two set of IDs, 10 digit ones and 20 digit ones. The IDs match Steamless Times (20 digits) and SDH-PlayTime (10 digit ones).

Every single non-Steam item from my library has both 10 digit number and 20 digit one in that file. Only those 3 games that were not matched on import have no 10 digit ID, but have 20 digit one. Not sure what happened and why but I suppose it is related to the import issue. Also not sure how widespread it is.

**Pull request merged**